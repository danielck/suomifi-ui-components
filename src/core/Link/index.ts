export { Link, LinkProps } from './Link/Link';
export { ExternalLink, ExternalLinkProps } from './ExternalLink/ExternalLink';
export { SkipLink, SkipLinkProps } from './SkipLink/SkipLink';
export { RouterLink, RouterLinkProps } from './RouterLink/RouterLink';
export { LinkListItem, LinkListItemProps } from './LinkListItem/LinkListItem';
export { LinkList, LinkListProps } from './LinkList/LinkList';
