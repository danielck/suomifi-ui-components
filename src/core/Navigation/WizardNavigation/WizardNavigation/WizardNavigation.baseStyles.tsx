import { font } from '../../../theme/reset';
import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';

export const baseStyles = (theme: SuomifiTheme) => css`
  &.fi-wizard-navigation {
    padding: ${theme.spacing.s};

    &--small-screen {
      background: ${theme.colors.highlightLight3};
      .fi-wizard-navigation_heading {
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        ${font(theme)('heading4')}

        .fi-icon {
          color: ${theme.colors.highlightBase};
          width: 20px;
          height: 20px;
        }
      }
    }

    .fi-wizard-navigation_heading {
      ${font(theme)('heading4')}
    }
    .fi-wizard-navigation_list {
      border-top: 1px solid ${theme.colors.depthBase};
      list-style-type: none;
      margin: 0;
      padding: 0;
      padding-top: ${theme.spacing.s};
      margin-top: ${theme.spacing.s};
    }
  }
`;
