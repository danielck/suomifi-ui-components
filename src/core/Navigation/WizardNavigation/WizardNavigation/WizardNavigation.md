### Basic usage

A `<WizardNavigationItem>` can have one of these 6 statuses:

- default: reachable & incomplete step
- current: currently active step
- completed: a step where the user has filled all necessary information
- current-completed: a combination of current and completed statuses
- coming: a step which is unreachable at the moment (but will become available when e.g. the previous steps have been completed)
- disabled: disabled step, not reachable. We do not recommend the use of this status unless absolutely necessary

Please use `<RouterLink>` as the child of `<WizardNavigationItem>` to get inteded CSS styles. `<RouterLink>` is polymorphic, and can be rendered as any component of your choice, for example React Router Link.

```js
import {
  WizardNavigation,
  WizardNavigationItem,
  RouterLink
} from 'suomifi-ui-components';

const Comp = (props) => {
  const { children, ...passProps } = props;
  return <div {...passProps}>{props.children}</div>;
};

<div style={{ width: '350px' }}>
  <WizardNavigation heading="Steps" aria-label="Main">
    <WizardNavigationItem status="completed">
      <RouterLink
        href="https://suomi.fi"
        aria-label="Step 1. This step is completed"
      >
        Step 1
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="default">
      <RouterLink href="#">Step 2</RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="current">
      <RouterLink aria-current="step" href="#">
        Step 3
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 4 with a long text that wraps to the second line like
        this
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 5
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="disabled">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 6
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink
        asComponent={Comp}
        role="button"
        aria-disabled
        tabIndex={0}
      >
        Step 7
      </RouterLink>
    </WizardNavigationItem>
  </WizardNavigation>
</div>;
```

For frameworks where its internal link component is used as a wrapper for the actual link, for example NextJS, the following approach can be used:

```jsx static
<WizardNavigationItem status="default">
  <NextJSLink href="/some-route" passHref>
    <RouterLink>Step 1</RouterLink>
  </NextJSLink>
</WizardNavigationItem>
```

### Small screen version

```js
import {
  WizardNavigation,
  WizardNavigationItem,
  RouterLink
} from 'suomifi-ui-components';

const Comp = (props) => {
  const { children, ...passProps } = props;
  return <div {...passProps}>{props.children}</div>;
};

<div style={{ width: '300px' }}>
  <WizardNavigation
    variant="smallScreen"
    heading="Steps"
    initiallyExpanded={false}
    aria-label="Main"
  >
    <WizardNavigationItem status="default">
      <RouterLink href="https://suomi.fi">Step 1</RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="completed">
      <RouterLink
        href="#"
        aria-label="Step 2. This step is completed"
      >
        Step 2
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="current">
      <RouterLink aria-current="step" href="#">
        Step 3
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 4
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 5
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="disabled">
      <RouterLink aria-disabled role="link" tabIndex={0}>
        Step 6
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem status="coming">
      <RouterLink
        asComponent={Comp}
        role="button"
        aria-disabled
        tabIndex={0}
      >
        Step 7
      </RouterLink>
    </WizardNavigationItem>
  </WizardNavigation>
</div>;
```
