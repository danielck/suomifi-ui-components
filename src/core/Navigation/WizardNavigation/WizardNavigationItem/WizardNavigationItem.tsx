import React, { ReactNode } from 'react';
import classnames from 'classnames';
import { HtmlDiv, HtmlLi, HtmlSpan } from '../../../../reset';
import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../../theme';
import { baseStyles } from './WizardNavigationItem.baseStyles';
import styled from 'styled-components';

export interface WizardNavigationItemProps {
  /** Custom class */
  className?: string;
  /** Use the polymorphic RouterLink component as child to get intended CSS styling */
  children: ReactNode;
  /** Indicates the step's number */
  stepNumber: Number;
  /** Status of the item. Changes styling and availability */
  status: 'visited' | 'current' | 'proceed' | 'not-visited' | 'disabled';
}

const baseClassName = 'fi-wizard-navigation-item';
const visitedClassName = `${baseClassName}--visited`;
const currentClassName = `${baseClassName}--current`;
const proceedClassName = `${baseClassName}--proceed`;
const notVisitedClassName = `${baseClassName}--not-visited`;
const disabledClassName = `${baseClassName}--disabled`;

const innerWrapperClassName = `${baseClassName}_inner-wrapper`;
const stepNumberClassName = `${baseClassName}_step-number`;

const BaseWizardNavigationItem = ({
  className,
  children,
  status,
  stepNumber,
  ...passProps
}: WizardNavigationItemProps) => (
  <HtmlLi
    className={classnames(className, baseClassName, {
      [visitedClassName]: status === 'visited',
      [currentClassName]: status === 'current',
      [proceedClassName]: status === 'proceed',
      [notVisitedClassName]: status === 'not-visited',
      [disabledClassName]: status === 'disabled',
    })}
    aria-disabled={status === 'disabled' ? true : undefined}
    {...passProps}
  >
    <HtmlDiv className={innerWrapperClassName}>
      <HtmlSpan className={stepNumberClassName}>{stepNumber}</HtmlSpan>
      {children}
    </HtmlDiv>
  </HtmlLi>
);

const StyledWizardNavigationItem = styled(
  (props: WizardNavigationItemProps & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseWizardNavigationItem {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)}
`;

const WizardNavigationItem = (props: WizardNavigationItemProps) => (
  <SuomifiThemeConsumer>
    {({ suomifiTheme }) => (
      <StyledWizardNavigationItem theme={suomifiTheme} {...props} />
    )}
  </SuomifiThemeConsumer>
);

WizardNavigationItem.displayName = 'WizardNavigationItem';
export { WizardNavigationItem };
