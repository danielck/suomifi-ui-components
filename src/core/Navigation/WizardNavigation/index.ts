export {
  WizardNavigation,
  WizardNavigationProps,
} from './WizardNavigation/WizardNavigation';
export {
  WizardNavigationItem,
  WizardNavigationItemProps,
} from './WizardNavigationItem/WizardNavigationItem';
