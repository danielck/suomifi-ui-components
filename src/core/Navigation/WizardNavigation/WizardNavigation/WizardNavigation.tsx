import React, { forwardRef, ReactNode, useState } from 'react';
import { HtmlButton, HtmlDiv, HtmlNav, HtmlUl } from '../../../../reset';
import styled from 'styled-components';
import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../../theme';
import { baseStyles } from './WizardNavigation.baseStyles';
import classnames from 'classnames';
import { Icon } from '../../../Icon/Icon';
import { getConditionalAriaProp } from '../../../../utils/aria';

export interface WizardNavigationProps {
  /** Use the `<WizardNavigationItem>` components as children */
  children: ReactNode;
  /** Name for the navigation element. Don't use the word "navigation" since it will be read by screen reader regardless. */
  'aria-label': string;
  /** ID for the HTML nav element */
  id?: string;
  /**
   * Normal or small screen variant
   * @default normal
   */
  variant?: 'default' | 'smallScreen';
  /**
   * Initial expand status for the menu. Only applies to smallScreen variant
   * @default true
   */
  initiallyExpanded?: boolean;
  /** Navigation component heading text */
  heading: string;
  /** Custom classname to extend or customize */
  className?: string;
  /** Ref is forwarded to nav element. Alternative for React `ref` attribute. */
  forwardedRef?: React.Ref<HTMLElement>;
}

const baseClassName = 'fi-wizard-navigation';
const smallScreenClassName = `${baseClassName}--small-screen`;
const dividerClassName = `${baseClassName}_divider`;
const listClassName = `${baseClassName}_list`;
const headingClassName = `${baseClassName}_heading`;

const BaseWizardNavigation = ({
  children,
  'aria-label': ariaLabel,
  id,
  variant,
  initiallyExpanded,
  heading,
  className,
  forwardedRef,
}: WizardNavigationProps) => {
  const initiallyExpandedValue =
    initiallyExpanded !== undefined ? initiallyExpanded : true;
  const [smallScreenNavOpen, setSmallScreenNavOpen] = useState(
    initiallyExpandedValue,
  );

  return (
    <HtmlDiv
      className={classnames(baseClassName, className, {
        [smallScreenClassName]: variant === 'smallScreen',
      })}
    >
      {variant === 'smallScreen' ? (
        <HtmlButton
          className={headingClassName}
          onClick={() => setSmallScreenNavOpen(!smallScreenNavOpen)}
          aria-expanded={smallScreenNavOpen}
        >
          {heading}
          <Icon icon={smallScreenNavOpen ? 'chevronDown' : 'chevronRight'} />
        </HtmlButton>
      ) : (
        <HtmlDiv className={headingClassName}>{heading}</HtmlDiv>
      )}
      {((variant === 'smallScreen' && smallScreenNavOpen) ||
        variant !== 'smallScreen') && (
        <HtmlNav
          forwardedRef={forwardedRef}
          id={id}
          {...getConditionalAriaProp('aria-label', [ariaLabel])}
        >
          <HtmlDiv className={dividerClassName} />
          <HtmlUl className={listClassName}>{children}</HtmlUl>
        </HtmlNav>
      )}
    </HtmlDiv>
  );
};

const StyledWizardNavigation = styled(
  (props: WizardNavigationProps & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseWizardNavigation {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)}
`;

const WizardNavigation = forwardRef(
  (props: WizardNavigationProps, ref: React.Ref<HTMLElement>) => (
    <SuomifiThemeConsumer>
      {({ suomifiTheme }) => (
        <StyledWizardNavigation
          theme={suomifiTheme}
          forwardedRef={ref}
          {...props}
        />
      )}
    </SuomifiThemeConsumer>
  ),
);

WizardNavigation.displayName = 'WizardNavigation';
export { WizardNavigation };
