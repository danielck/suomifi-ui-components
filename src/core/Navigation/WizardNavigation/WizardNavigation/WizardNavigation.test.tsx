import React from 'react';
import { render } from '@testing-library/react';
import { axeTest } from '../../../../utils/test';

import { WizardNavigation } from './WizardNavigation';
import { WizardNavigationItem } from '../WizardNavigationItem/WizardNavigationItem';
import { RouterLink } from '../../../Link';

const TestWizardNavigation = (
  <WizardNavigation heading="Steps">
    <WizardNavigationItem stepNumber={1} status="visited">
      <RouterLink href="https://suomi.fi">Step 1 (visited)</RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem stepNumber={2} status="current">
      <RouterLink aria-current="step" aria-disabled href="#">
        Step 2 (current)
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem stepNumber={3} status="proceed">
      <RouterLink
        asComponent="button"
        onClick={() => console.log('Step 3 clicked!')}
      >
        Step 3 (proceed)
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem stepNumber={4} status="not-visited">
      <RouterLink aria-disabled tabIndex={-1} href="#">
        Step 4 (not visited)
      </RouterLink>
    </WizardNavigationItem>
    <WizardNavigationItem stepNumber={5} status="disabled">
      <RouterLink aria-disabled tabIndex={-1} href="#">
        Step 5 (disabled)
      </RouterLink>
    </WizardNavigationItem>
  </WizardNavigation>
);

test('calling render with the same component on the same container does not remount', () => {
  const WizardNavRendered = render(TestWizardNavigation);
  const { container } = WizardNavRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test(
  'should not have basic accessibility issues',
  axeTest(TestWizardNavigation),
);
