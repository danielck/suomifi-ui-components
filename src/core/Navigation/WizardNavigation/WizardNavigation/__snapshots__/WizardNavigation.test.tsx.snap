// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c7 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: transparent;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7::before,
.c7::after {
  box-sizing: border-box;
}

.c7:hover,
.c7:active,
.c7:focus,
.c7:focus-within {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: list-item;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c6 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c6::before,
.c6::after {
  box-sizing: border-box;
}

.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  list-style-type: decimal;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 0;
  margin-right: 0;
  padding-left: 40px;
}

.c3::before,
.c3::after {
  box-sizing: border-box;
}

.c1.fi-wizard-navigation {
  padding: 15px;
}

.c1.fi-wizard-navigation--small-screen {
  background: hsl(212,63%,95%);
}

.c1.fi-wizard-navigation--small-screen .fi-wizard-navigation_heading {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
}

.c1.fi-wizard-navigation--small-screen .fi-wizard-navigation_heading .fi-icon {
  color: hsl(212,63%,45%);
  width: 20px;
  height: 20px;
}

.c1.fi-wizard-navigation .fi-wizard-navigation_heading {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
}

.c1.fi-wizard-navigation .fi-wizard-navigation_list {
  border-top: 1px solid hsl(202,7%,67%);
  list-style-type: none;
  margin: 0;
  padding: 0;
  padding-top: 15px;
  margin-top: 15px;
}

.c5.fi-wizard-navigation-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c5.fi-wizard-navigation-item:not(:last-child) {
  margin-bottom: 22px;
}

.c5.fi-wizard-navigation-item:not(:last-child):after {
  content: '';
  height: 10px;
  width: 2px;
  background: hsl(202,7%,67%);
  position: absolute;
  bottom: -16px;
  left: 12px;
}

.c5.fi-wizard-navigation-item .fi-wizard-navigation-item_inner-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5.fi-wizard-navigation-item .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  border: 1px solid hsl(202,7%,40%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 25px;
  height: 25px;
  font-weight: 600;
  line-height: 27px;
  font-size: 18px;
  margin-right: 10px;
}

.c5.fi-wizard-navigation-item .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  margin-bottom: 1px;
}

.c5.fi-wizard-navigation-item--visited .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  background: hsl(165,90%,27%);
  color: hsl(0,0%,100%);
}

.c5.fi-wizard-navigation-item--visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  cursor: pointer;
}

.c5.fi-wizard-navigation-item--visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5.fi-wizard-navigation-item--visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router:visited {
  color: hsl(212,63%,45%);
}

.c5.fi-wizard-navigation-item--current .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  background: hsl(165,90%,27%);
  color: hsl(0,0%,100%);
}

.c5.fi-wizard-navigation-item--current .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  pointer-events: none;
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  margin-bottom: 1px;
}

.c5.fi-wizard-navigation-item--current .fi-wizard-navigation-item_inner-wrapper .fi-link--router:hover,
.c5.fi-wizard-navigation-item--current .fi-wizard-navigation-item_inner-wrapper .fi-link--router:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,16%);
}

.c5.fi-wizard-navigation-item--proceed .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  background: hsl(0,0%,100%);
  color: hsl(202,7%,40%);
  border: 1px solid hsl(166,90%,34%);
}

.c5.fi-wizard-navigation-item--proceed .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  cursor: pointer;
}

.c5.fi-wizard-navigation-item--proceed .fi-wizard-navigation-item_inner-wrapper .fi-link--router:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5.fi-wizard-navigation-item--proceed .fi-wizard-navigation-item_inner-wrapper .fi-link--router:visited {
  color: hsl(212,63%,45%);
}

.c5.fi-wizard-navigation-item--not-visited .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  background: hsl(0,0%,100%);
  color: hsl(0,0%,16%);
  border: 1px solid hsl(0,0%,16%);
}

.c5.fi-wizard-navigation-item--not-visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  pointer-events: none;
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-wizard-navigation-item--not-visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5.fi-wizard-navigation-item--not-visited .fi-wizard-navigation-item_inner-wrapper .fi-link--router:visited {
  color: hsl(0,0%,16%);
}

.c5.fi-wizard-navigation-item--disabled .fi-wizard-navigation-item_inner-wrapper .fi-wizard-navigation-item_step-number {
  background: hsl(0,0%,100%);
  color: hsl(202,7%,40%);
  border: 1px solid hsl(202,7%,80%);
}

.c5.fi-wizard-navigation-item--disabled .fi-wizard-navigation-item_inner-wrapper .fi-link--router {
  pointer-events: none;
  color: hsl(202,7%,67%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-wizard-navigation-item--disabled .fi-wizard-navigation-item_inner-wrapper .fi-link--router:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5.fi-wizard-navigation-item--disabled .fi-wizard-navigation-item_inner-wrapper .fi-link--router:visited {
  color: hsl(202,7%,67%);
}

.c8 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c8.fi-link--router {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8.fi-link--router:hover,
.c8.fi-link--router:active,
.c8.fi-link--router:focus,
.c8.fi-link--router:focus-within {
  color: hsl(212,63%,45%);
}

.c8.fi-link--router:focus,
.c8.fi-link--router:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8.fi-link--router:focus {
  outline: 0;
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
}

.c8.fi-link--router:hover,
.c8.fi-link--router:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8.fi-link--router:visited {
  color: hsl(284,36%,45%);
}

<div
  class="c0 fi-wizard-navigation c1"
>
  <div
    class="c0 fi-wizard-navigation_heading"
  >
    Steps
  </div>
  <nav
    class="c2"
  >
    <ul
      class="c3 fi-wizard-navigation_list"
    >
      <li
        class="c4 c5 fi-wizard-navigation-item fi-wizard-navigation-item--visited"
      >
        <div
          class="c0 fi-wizard-navigation-item_inner-wrapper"
        >
          <span
            class="c6 fi-wizard-navigation-item_step-number"
          >
            1
          </span>
          <a
            class="c7 fi-link--router c8"
            href="https://suomi.fi"
          >
            Step 1 (visited)
          </a>
        </div>
      </li>
      <li
        class="c4 c5 fi-wizard-navigation-item fi-wizard-navigation-item--current"
      >
        <div
          class="c0 fi-wizard-navigation-item_inner-wrapper"
        >
          <span
            class="c6 fi-wizard-navigation-item_step-number"
          >
            2
          </span>
          <a
            aria-current="step"
            aria-disabled="true"
            class="c7 fi-link--router c8"
            href="#"
          >
            Step 2 (current)
          </a>
        </div>
      </li>
      <li
        class="c4 c5 fi-wizard-navigation-item fi-wizard-navigation-item--proceed"
      >
        <div
          class="c0 fi-wizard-navigation-item_inner-wrapper"
        >
          <span
            class="c6 fi-wizard-navigation-item_step-number"
          >
            3
          </span>
          <button
            class="fi-link--router c8"
          >
            Step 3 (proceed)
          </button>
        </div>
      </li>
      <li
        class="c4 c5 fi-wizard-navigation-item fi-wizard-navigation-item--not-visited"
      >
        <div
          class="c0 fi-wizard-navigation-item_inner-wrapper"
        >
          <span
            class="c6 fi-wizard-navigation-item_step-number"
          >
            4
          </span>
          <a
            aria-disabled="true"
            class="c7 fi-link--router c8"
            href="#"
            tabindex="-1"
          >
            Step 4 (not visited)
          </a>
        </div>
      </li>
      <li
        aria-disabled="true"
        class="c4 c5 fi-wizard-navigation-item fi-wizard-navigation-item--disabled"
      >
        <div
          class="c0 fi-wizard-navigation-item_inner-wrapper"
        >
          <span
            class="c6 fi-wizard-navigation-item_step-number"
          >
            5
          </span>
          <a
            aria-disabled="true"
            class="c7 fi-link--router c8"
            href="#"
            tabindex="-1"
          >
            Step 5 (disabled)
          </a>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;
